common/src/java/org/apache/hadoop/hive/common/FileUtils.java

```diff
@@ -926,8 +925,7 @@ public static File createLocalDirsTempFile(Configuration conf, String prefix, St
    * delete a temporary file and remove it from delete-on-exit hook.
    */
   public static boolean deleteTmpFile(File tempFile) {
-    if (tempFile != null) {
-      tempFile.delete();
+    if (tempFile != null && tempFile.delete()) {
```

Comment:
this change will leave the file registered in the shutdownmanager in case the file was already deleted

Reply:
Good catch, fixed